core/utils/index_utils.cpp

```diff
@@ -28,149 +28,27 @@
 
 #include "formats/format_utils.hpp"
 
-namespace {
-
-// Returns percentage of live documents
-inline double FillFactor(const irs::SegmentInfo& segment) noexcept {
-  return static_cast<double>(segment.live_docs_count) /
-         static_cast<double>(segment.docs_count);
-}
-
-// Returns approximated size of a segment in the absence of removals
-inline size_t SizeWithoutRemovals(const irs::SegmentInfo& segment) noexcept {
-  return size_t(static_cast<double>(segment.byte_size) * FillFactor(segment));
-}
-
 namespace tier {
 
-struct SegmentStats {
-  // cppcheck-suppress noExplicitConstructor
-  SegmentStats(const irs::SubReader& reader) noexcept
-    : reader{&reader},
-      meta{&reader.Meta()},
-      size{SizeWithoutRemovals(*meta)},
-      fill_factor{FillFactor(*meta)} {}
-
-  bool operator<(const SegmentStats& rhs) const noexcept {
-    // cppcheck-suppress constVariable
-    auto& lhs = *this;
-
-    if (lhs.size == rhs.size) {
-      if (lhs.fill_factor > rhs.fill_factor) {
-        return true;
-      } else if (lhs.fill_factor < rhs.fill_factor) {
-        return false;
-      }
-
-      return lhs.meta->name < rhs.meta->name;
+    //  interface to fetch the required attributes from
+    //  SegmentStats struct.
+    //  We use this function in struct ConsolidationCandidate
+    //  to fetch the segment dimensions from the SegmentStats
+    //  struct.
+    //
+    void getSegmentDimensions(
+        std::vector<tier::SegmentStats>::const_iterator itr,
+        uint64_t& byte_size,
+        uint64_t& docs_count,
+        uint64_t& live_docs_count) {
+
+      auto itrMeta = itr->meta;
```

Comment:
Personally, I have a slight preference towards
```suggestion
      auto* itrMeta = itr->meta;
```
, but feel free to keep it as is if you prefer it that way.

Reply:
`auto *` definitely is more appropriate since we are expecting itr->meta always to be a pointer.
Changed it.